Quick_Deploy/vLLM/README.md

```diff
@@ -0,0 +1,147 @@
+that needs to be performed in Triton for more streamline integration with vLLM. 
+These implementations can change in future for a better developer experience.
+
+This README showcases how to deploy a simple [facebook/opt-125m](https://huggingface.co/facebook/opt-125m) model on Triton Inference Server with [vLLM](https://github.com/vllm-project/vllm). We will be using Triton's [Python backend](https://github.com/triton-inference-server/python_backend) to host vLLM integration.
+
+## Step 1: Build a Triton Container Image with vLLM
+
+vLLM with all its dependencies is quite large hence it is not feasible to ship these dependencies in a [custom environment](https://github.com/triton-inference-server/python_backend#creating-custom-execution-environments).
+We will build a new container image derived from tritonserver:22.12-py3 with all the dependencies needed to run the model.
+The dependencies that will be installed in the container image are listed in requirements.txt
+
+```
+docker build -t tritonserver_vllm:22.12-py3 .
+```
+
+The above command should create tritonserver_vllm:22.12-py3 image with all the dependencies.
+
+## Step 2: Set Up Triton Inference Server
+
+To use Triton, we need to build a model repository. The structure of the repository as follows:
+```
+model_repository
+|
++-- vllm
+    |
+    +-- config.pbtxt
+    +-- vllm_engine_args.json
+    +-- 1
+        |
+        +-- model.py
+```
+
+A sample model repository for deploying `facebook/opt-125m` using vLLM in Triton is included with this demo as `model_repository` directory. The content of `vllm_engine_args.json` is:
+
+```json
+{
+    "model":"facebook/opt-125m"
+}
+```
+This file can be modified to provide further settings to the vLLM engine. Look at VLLMAsyncEngineConfig in [model.py](model_repository/vllm/1/model.py) for supported fields.
+
+Read through the documentation in [`config.pbtxt`](model_repository/vllm/config.pbtxt) and [`model.py`](model_repository/vllm/1/model.py) to
+understand how to configure this sample for your use-case.
+
+
+```
+docker run --gpus all --rm --shm-size=1G --ulimit memlock=-1 --ulimit stack=67108864 -p 8000:8000 -p 8001:8001 -p 8002:8002 -v ${PWD}/model_repository:/models tritonserver_vllm:22.12-py3 tritonserver --model-repository=/models
+```
+
+
+
+## Step 3: Using a Triton Client to Query the Server
+
+We will run the client within Triton's SDK container to issue multiple async requests using
+[gRPC asyncio client](https://github.com/triton-inference-server/client/blob/main/src/python/library/tritonclient/grpc/aio/__init__.py)
+library.
+
+```
+docker run -it --net=host -v ${PWD}:/workspace/ nvcr.io/nvidia/tritonserver:22.12-py3-sdk bash
+```
+
+Within the container, run the [`client.py`](client.py) as:
+
+```
+python3 client.py
+
+```
+
+The output of the client should look like below:
+```
+===========
+prompt => 'T h e   f u t u r e   o f   A I   i s'
```

Comment:
I was wondering what is the reason for spaces between the prompt letters?

Reply:
Fixed. There was an addition " ".join call which I forgot to remove...